libraries/lib-numeric-formats/NumericConverterFormatter.cpp

```diff
@@ -0,0 +1,73 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/**********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file NumericConverterFormatter.cpp
+
+ Dmitry Vedenko
+
+ **********************************************************************/
+#include "NumericConverterFormatter.h"
+
+#include <cmath>
+
+namespace
+{
+size_t CalculateDigits(size_t rangeEnd)
+{
+   if (rangeEnd == 0)
+      return 0;
+
+   --rangeEnd;
+
+   size_t digitsCount = 0;
+
+   while (rangeEnd > 0)
+   {
+      rangeEnd /= 10;
+      ++digitsCount;
+   }
+
+   return digitsCount;
+}
+}
+
+NumericField::NumericField(size_t _digits, bool zeropad)
+    : digits { _digits }
+{
+   if (zeropad && digits > 1)
+      formatStr.Printf(wxT("%%0%zud"), digits); // ex. "%03d" if digits is 3
+   else
+      formatStr = "%d";
```

Comment:
I notice that the behavior of the false branch changed, and documentation doesn't explain it, but then too, it is never reached anyway because the second argument is always true.